
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://seqscope.github.io/NovaScope-exemplary-downstream-analysis/analysis/hex_idx/step2b-seurat/">
      
      
        <link rel="prev" href="../step2a-LDA/">
      
      
        <link rel="next" href="../step3-transform/">
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.13">
    
    
      
        <title>Step2b. Seurat Clustering - NovaScope-exemplary-downstream-analysis</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.7e359304.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../css/custom.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#step-2b-inferring-cell-type-factors-using-seurat" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="NovaScope-exemplary-downstream-analysis" class="md-header__button md-logo" aria-label="NovaScope-exemplary-downstream-analysis" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NovaScope-exemplary-downstream-analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Step2b. Seurat Clustering
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8v2m9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1 0 1.71-1.39 3.1-3.1 3.1h-4V17h4a5 5 0 0 0 5-5 5 5 0 0 0-5-5Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5M7 15a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/seqscope/NovaScope-exemplary-downstream-analysis" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    seqscope/NEDA
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../.." class="md-tabs__link">
          
  
    
  
  Home

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../installation/installation/" class="md-tabs__link">
          
  
    
  
  Installation

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../intro/" class="md-tabs__link">
          
  
    
  
  Pixel-level Analysis

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../cell_idx/intro/" class="md-tabs__link">
          
  
    
  
  Cell Segmentation-based Analysis

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="NovaScope-exemplary-downstream-analysis" class="md-nav__button md-logo" aria-label="NovaScope-exemplary-downstream-analysis" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    NovaScope-exemplary-downstream-analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/seqscope/NovaScope-exemplary-downstream-analysis" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    seqscope/NEDA
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
        
          
          
          <div class="md-nav__link md-nav__container">
            <a href="../../.." class="md-nav__link ">
              
  
  <span class="md-ellipsis">
    Home
  </span>
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_1" id="__nav_1_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Home
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../home/documentation_overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Documentation Overview
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Installation
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Installation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../installation/installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installing NEDA
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../installation/example_data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Example Datasets
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Pixel-level Analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Pixel-level Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
      
      
        
          
          
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Preparing Input
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Preparing Input
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../prepare_data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Input Data
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../job_config/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Input Configuration
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
      
      
        
          
          
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3_3">
            <span class="md-nav__icon md-icon"></span>
            Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../step1-preprocess/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Step1. Preprocessing
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../step2a-LDA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Step2a. LDA Factorization
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Step2b. Seurat Clustering
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Step2b. Seurat Clustering
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-2b1-data-evaluation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.1 Data Evaluation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b2-manually-selecting-the-optimal-nfeature_rna-cutoff" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.2 Manually Selecting the Optimal nFeature_RNA Cutoff
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b3-seurat-clustering-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.3 Seurat Clustering Analysis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b4-manually-selecting-the-optimal-clustering-resolution" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.4 Manually Selecting the Optimal Clustering Resolution
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b5-preparing-a-count-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.5 Preparing a Count Matrix
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../step3-transform/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Step3. Transform
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../step4-decode/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Step4. Pixel-level Decoding
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Cell Segmentation-based Analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Cell Segmentation-based Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cell_idx/intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cell_idx/prepare_data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Input Data
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
        
          
          <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_3">
            <span class="md-nav__icon md-icon"></span>
            Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cell_idx/step1-cell_SGE/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Step1. Create Cell-indexed SGE
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cell_idx/step2-Seurat-clustering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Step2. Seurat Clustering
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-2b1-data-evaluation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.1 Data Evaluation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b2-manually-selecting-the-optimal-nfeature_rna-cutoff" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.2 Manually Selecting the Optimal nFeature_RNA Cutoff
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b3-seurat-clustering-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.3 Seurat Clustering Analysis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b4-manually-selecting-the-optimal-clustering-resolution" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.4 Manually Selecting the Optimal Clustering Resolution
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2b5-preparing-a-count-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2b.5 Preparing a Count Matrix
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="step-2b-inferring-cell-type-factors-using-seurat">Step 2b. Inferring Cell Type Factors using Seurat<a class="headerlink" href="#step-2b-inferring-cell-type-factors-using-seurat" title="Permanent link">&para;</a></h1>
<p>This example illustrates infering cell type factors using Seurat. This process contains two stops that require manual evaluations at <a href="#step-2b2-manually-select-the-cutoffs">step 2b.2</a> and <a href="#step-2b4-manually-select-the-resolution-for-clustering">step 2b.4</a>.</p>
<p><strong>Prefix</strong>:</p>
<p>This documentation uses the following prefixes to illustrate input and output filenames, which are automatically assigned by the script.</p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">train_prefix</span><span class="o">=</span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">prefix</span><span class="si">}</span><span class="s2">.</span><span class="si">${</span><span class="nv">solo_feature</span><span class="si">}</span><span class="s2">.nf</span><span class="si">${</span><span class="nv">nfactor</span><span class="si">}</span><span class="s2">.d_</span><span class="si">${</span><span class="nv">train_width</span><span class="si">}</span><span class="s2">.s_</span><span class="si">${</span><span class="nv">train_n_epoch</span><span class="si">}</span><span class="s2">&quot;</span>
</code></pre></div></td></tr></table></div>
<ul>
<li>The <code>nfactor</code> will be determined in <a href="#step-2b5-prepare-a-count-matrix-with-the-selected-resolution">step 2b.5</a>.</li>
<li>Variable details for the prefixes are in the <a href="../job_config/">Job Configuration</a>.</li>
</ul>
<h2 id="step-2b1-data-evaluation">Step 2b.1 Data Evaluation<a class="headerlink" href="#step-2b1-data-evaluation" title="Permanent link">&para;</a></h2>
<p>This step applies the <code>Seurat_analysis.R</code> script in test mode to</p>
<ul>
<li>assess counts of mitochondrial and hypothetical genes,</li>
<li>examine the distribution of the number of spatial barcodes per hexagon (nCount_RNA) and the number of genes detected per hexagon (<code>Nfeature_RNA</code>),</li>
<li>evaluate assess the performance of different nFeature_RNA thresholds — 50, 100, 200, 300, 400, 500, 750, and 1000 — to filter the input hexagon-indexed SGE.</li>
</ul>
<p>Input &amp; Output
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span>
<span class="normal">5</span>
<span class="normal">6</span>
<span class="normal">7</span>
<span class="normal">8</span>
<span class="normal">9</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="c1"># Input: </span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/barcodes.tsv.gz<span class="w">                        </span><span class="c1">## user-defined input hexagon SGE in 10X Genomics format</span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/features.tsv.gz<span class="w">     </span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/matrix.mtx.gz<span class="w">  </span>

<span class="c1"># Output: </span>
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/Ncount_Nfeature_vln.png
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/nFeature_RNA_dist.png
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/nFeature_RNA_cutoff<span class="si">${</span><span class="nv">cutoff</span><span class="si">}</span>.png<span class="w">       </span><span class="c1">## a density plot with two panels, displaying the hexagon-indexed SGE before and after filtering by Nfeature_RNA, for each nFeature_RNA cutoff</span>
</code></pre></div></td></tr></table></div></p>
<p>Commands:
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">$neda_dir</span>/steps/step2b.1-Seurat-test-cutoff.sh<span class="w"> </span><span class="nv">$input_configfile</span>
</code></pre></div></td></tr></table></div></p>
<h2 id="step-2b2-manually-selecting-the-optimal-nfeature_rna-cutoff">Step 2b.2 Manually Selecting the Optimal nFeature_RNA Cutoff<a class="headerlink" href="#step-2b2-manually-selecting-the-optimal-nfeature_rna-cutoff" title="Permanent link">&para;</a></h2>
<p>Examine density plots generated by <a href="#step-2b1-data-evaluation">step 2b.1</a> to choose the optimal <code>nFeature_RNA</code> cutoff to remove noise signals. For our example data, we applied a cut off of 500 for the deep liver section dataset, and 100 for the shallow liver section dataset and the minimal test dataset. It is optional to define x y ranges(<code>X_min</code>, <code>X_max</code>, <code>Y_min</code>, and <code>Y_max</code>).</p>
<p>Add the following variables to the input configuration file.</p>
<p>Example:
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span>
<span class="normal">5</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">nFeature_RNA_cutoff</span><span class="o">=</span><span class="m">100</span><span class="w">   </span>
<span class="nv">X_min</span><span class="o">=</span><span class="m">2</span>.5e+06
<span class="nv">X_max</span><span class="o">=</span>1e+07
<span class="nv">Y_min</span><span class="o">=</span>1e+06
<span class="nv">Y_max</span><span class="o">=</span>6e+06
</code></pre></div></td></tr></table></div></p>
<h2 id="step-2b3-seurat-clustering-analysis">Step 2b.3 Seurat Clustering Analysis<a class="headerlink" href="#step-2b3-seurat-clustering-analysis" title="Permanent link">&para;</a></h2>
<p>This step starts with removing mitochondrial and hypothetical genes and filtering the hexagon-indexed SGE matrix by <code>nFeature_RNA_cutoff</code>. When X Y coordinate ranges are applied, the hexagon-indexed SGE matrix will also be filtered by coordinates. </p>
<p>Subsequently, this step applies <a href="https://github.com/satijalab/sctransform">sctransform</a> normalization followed by dimensionality reduction through <a href="https://satijalab.org/seurat/reference/runpca">Principal Component Analysis (PCA)</a> and <a href="https://satijalab.org/seurat/reference/runumap">Uniform Manifold Approximation and Projection (UMAP)</a> embedding.</p>
<p>Next, the step employs <a href="https://satijalab.org/seurat/reference/findclusters"><code>FindClusters</code></a> to segregate hexagons into clusters utilizing a shared nearest neighbor (SNN) modularity optimization-based clustering algorithm. During this process, <a href="https://satijalab.org/seurat/reference/findclusters"><code>FindClusters</code></a> applies an argument of <code>resolution</code> to determine the "granularity" of clusters, i.e., a higher resolution value yields more clusters. A range of <code>resolutions</code>, including 0.25, 0.5, 0.75, 1, 1.25, 1.5, 1.75, and 2, are applied to explore the optimal <code>resolutions</code>. For each resolution, the step generates an UMAP for dimensionality reduction, a spatial plot to visualize the clusters and their spatial arrangement, and a CSV file of differentially expressed genes for each cluster.</p>
<p>Additionally, this step generates a metadata file containing information on the cluster assignment for each hexagon, and an RDS (R Data Serialization) file that stores the complete Seurat object with all the compiled data.</p>
<p>Input &amp; Output
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"> 1</span>
<span class="normal"> 2</span>
<span class="normal"> 3</span>
<span class="normal"> 4</span>
<span class="normal"> 5</span>
<span class="normal"> 6</span>
<span class="normal"> 7</span>
<span class="normal"> 8</span>
<span class="normal"> 9</span>
<span class="normal">10</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="c1"># Input: </span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/barcodes.tsv.gz<span class="w">                                                    </span><span class="c1">## user-defined input hexagon-indexed SGE in 10X Genomics-compatible format</span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/features.tsv.gz<span class="w">     </span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/matrix.mtx.gz<span class="w">     </span>

<span class="c1"># Output: </span>
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_cutoff<span class="si">${</span><span class="nv">nFeature_RNA_cutoff</span><span class="si">}</span>_metadata.csv<span class="w">                </span><span class="c1">## a metadata file</span>
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_cutoff<span class="si">${</span><span class="nv">nFeature_RNA_cutoff</span><span class="si">}</span>_SCT.RDS<span class="w">                     </span><span class="c1">## an RDS file</span>
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_cutoff<span class="si">${</span><span class="nv">nFeature_RNA_cutoff</span><span class="si">}</span>_res<span class="si">${</span><span class="nv">res</span><span class="si">}</span>_DE.csv<span class="w">            </span><span class="c1">## Each resolution returns a CSV file of differentially expressed genes for each cluster</span>
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_cutoff<span class="si">${</span><span class="nv">nFeature_RNA_cutoff</span><span class="si">}</span>_res<span class="si">${</span><span class="nv">res</span><span class="si">}</span>_DimSpatial.png<span class="w">    </span><span class="c1">## Each resolution returns an image of two panels including an UMAP for dimensionality reduction, a spatial plot to visualize the clusters and their spatial arrangement </span>
</code></pre></div></td></tr></table></div></p>
<p>Commands:
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">$neda_dir</span>/steps/step2b.3-Seurat-clustering.sh<span class="w"> </span><span class="nv">$input_configfile</span>
</code></pre></div></td></tr></table></div></p>
<h2 id="step-2b4-manually-selecting-the-optimal-clustering-resolution">Step 2b.4 Manually Selecting the Optimal Clustering Resolution<a class="headerlink" href="#step-2b4-manually-selecting-the-optimal-clustering-resolution" title="Permanent link">&para;</a></h2>
<p>Examine the UMAP and spatial plots from the <a href="#step-2b3-seurat-clustering-analysis">step 2b.3</a> and choose the optimal <code>resolution</code>. Then, save this chosen <code>resolution</code> into the input configuration file as the <code>res_of_interest</code> variable.</p>
<p>Example:
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">res_of_interest</span><span class="o">=</span><span class="m">1</span>
</code></pre></div></td></tr></table></div></p>
<h2 id="step-2b5-preparing-a-count-matrix">Step 2b.5 Preparing a Count Matrix<a class="headerlink" href="#step-2b5-preparing-a-count-matrix" title="Permanent link">&para;</a></h2>
<p>Transform the metadata file into a count matrix to serve as the model matrix for the subsequent steps. This step automatically detects the number of clusters from the model matrix and assignes it as a <code>nfactor</code> variable in the input configuration file.</p>
<p>Input &amp; Output
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span>
<span class="normal">5</span>
<span class="normal">6</span>
<span class="normal">7</span>
<span class="normal">8</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="c1"># Input:</span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/barcodes.tsv.gz<span class="w">                                                       </span><span class="c1">## user-defined input hexagon-indexed SGE in 10X Genomics-compatible format</span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/features.tsv.gz<span class="w">     </span>
<span class="si">${</span><span class="nv">input_hexagon_sge_10x_dir</span><span class="si">}</span>/matrix.mtx.gz<span class="w">     </span>
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_cutoff<span class="si">${</span><span class="nv">nFeature_RNA_cutoff</span><span class="si">}</span>_metadata.csv<span class="w">                   </span><span class="c1">## the meta file from step 2b.3</span>

<span class="c1"># Output: </span>
<span class="si">${</span><span class="nv">output_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_model</span><span class="si">}</span>/<span class="si">${</span><span class="nv">train_prefix</span><span class="si">}</span>.model_matrix.tsv.gz
</code></pre></div></td></tr></table></div></p>
<p>Commands:
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="nv">$neda_dir</span>/steps/step2b.5-Seurat-count-matrix.sh<span class="w"> </span><span class="nv">$input_configfile</span>
</code></pre></div></td></tr></table></div></p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.tabs", "navigation.indexes", "navigation.sections", "content.code.annotate"], "search": "../../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.c8d2eff1.min.js"></script>
      
        <script src="../../../js/open_in_new_tab.js"></script>
      
    
  </body>
</html>