# Seurat + FICTURE analytical strategy

## Step 2b. Infer Cell Type Factors using Seurat

This example illustrates infering cell type factors using Seurat. This process contains two stops that require manual evaluation, including one at [step 2b.2](#step-2b2-manually-select-the-cutoffs) and the other at [step 2b.4](#step-2b4-manually-select-the-resolution-for-clustering). 

**Prefix**:

This documentation uses the following prefixes to illustrate input and output filenames, which are automatically assigned by the script. 

```
hexagon_prefix="${prefix}.hexagon.${sf}.d_${tw}"
train_prefix="${prefix}.${sf}.nF${nf}.d_${tw}.s_${ep}"    
```

* The `nf` will be determined in [step 2b.5](#step-2b5-prepare-a-count-matrix-with-the-selected-resolution).
* Details on variables used in above prefixes are in the [Job Configuration](../../prep_input/job_config.md).

### Step 2b.1 Create Hexagonal Spatial Digital Gene Expression (SGE) Matrix and Test nFeature_RNA Cutoffs
This step creates hexagonal spatial gene expression (SGE) matrix that is compatible with Seurat. It analyzes the distribution of Ncount and Nfeature, and tests filtering the SGE using different nFeature_RNA cutoffs, including 50, 100, 200, 300, 400, 500, 750, and 1000. For each nFeature_RNA cutoff, a density plot is produced to effectively illustrate its efficacy.

Input & Output
```
# Input:
${output_dir}/${prefix}.merged.matrix.tsv.gz
${output_dir}/${prefix}.feature.tsv.gz

# Output: 
# * Hexagonal SGE: 
        ${output_dir}/${train_model}/features.tsv.gz
        ${output_dir}/${train_model}/barcodes.tsv.gz
        ${output_dir}/${train_model}/matrix.mtx.gz
# * Evaluation files: 
        ${output_dir}/${train_model}/Ncount_Nfeature_vln.png
        ${output_dir}/${train_model}/nFeature_RNA_dist.png
        # * for each cut off ${cutoff} in 50, 100, 200, 300, 400, 500, 750, and 1000:
                ${output_dir}/${train_model}/nFeature_RNA_cutoff${cutoff}.png
```

Command:
```bash
$neda_dir/steps/step2b.1-creat-hexagons-for-Seurat.sh $input_configfile
```

### Step 2b.2 Select the Optimal nFeature_RNA Cutoff

Examine the density plots generated by [step 2b.1](#step-2b1-create-hexagonal-spatial-gene-expression-sge-matrix-and-test-different-cutoffs-for-nfeature_rna) to choose a cutoff for nFeature_RNA, aiming to remove noise signals. For our example data, we applied a cut off of 500 for deep liver section dataset, and 100 for shallow liver section data. 

It is optional to define x y ranges. 

Add those variables to the `input_data_and_params` file.

Example:
```
# In this case, the Y_max is not applied. 
nFeature_RNA_cutoff=100
X_min=2.5e+06
X_max=1e+07
Y_min=1e+06
```

### Step 2b.3 Seurat Clustering Analysis
The `Seurat_analysis.R` script, by default, evaluates clustering at various resolutions, specifically 0.25, 0.5, 0.75, 1, 1.25, 1.5, and 1.75. 

For each resolution level, it creates a Dimensionality Reduction and Spatial plot to illustrate the clusters through Uniform Manifold Approximation and Projection (UMAP) along with their spatial distribution. It also produces a Differential Expression (DE) file that lists the marker genes for each cluster.

Additionally, this step generates a metadata file containing information on the cluster assignment for each hexagon, and an RDS file that stores the complete Seurat object with all the compiled data.

Input & Output
```
# Input: 
${output_dir}/${train_model}/features.tsv.gz
${output_dir}/${train_model}/barcodes.tsv.gz
${output_dir}/${train_model}/matrix.mtx.gz

#Output: 
# * A metadata file:
        ${output_dir}/${train_model}/${prefix}_cutoff${nFeature_RNA_cutoff}_metadata.csv
# * An RDS file:
        ${output_dir}/${train_model}/${prefix}_cutoff${nFeature_RNA_cutoff}_SCT.RDS
# * For each resolution (`$res`) in 0.25, 0.5, 0.75, 1, 1.25, 1.5, and 1.75:
        ${output_dir}/${train_model}/${prefix}_cutoff${nFeature_RNA_cutoff}_res${res}_DE.csv
        ${output_dir}/${train_model}/${prefix}_cutoff${nFeature_RNA_cutoff}_res${res}_DimSpatial.png
```

Command:
```bash
$neda_dir/steps/step2b.3-Seurat-clustering.sh $input_configfile
```

### Step 2b.4 Select the Optimal Clustering Resolution
Examine the Dimensionality Reduction and Spatial plots from the [step 2b.3](#step-2b3-seurat-clustering-analysis) and choose a resolution to continue with. Then, save this chosen resolution into the `input_data_and_params` file as the res_of_interest variable. 

Example:
```
res_of_interest=1
```

### Step 2b.5 Prepare a Count Matrix
Transform the metadata into a count matrix to serve as the model matrix for the subsequent step.  Additionally, the number of clusters from the model matrix will be obtained and assigned to the `nf` variable in the `input_data_and_params` file.

Input & Output
```
#Input:
${output_dir}/${train_model}/features.tsv.gz
${output_dir}/${train_model}/barcodes.tsv.gz
${output_dir}/${train_model}/matrix.mtx.gz
${output_dir}/${train_model}/${prefix}_cutoff${nFeature_RNA_cutoff}_metadata.csv

#Output: 
${output_dir}/${train_model}/${train_prefix}.model.tsv.gz
```

Command:
```bash
$neda_dir/steps/step2b.5-Seurat-count-matrix.sh $input_configfile
```
